### ISSUE TITLE
Is synthetic data from generative models ready for image recognition?

### ISSUE THREAD

**Issue by chaerinkong** [reyllama/paper-reviews#199]
# TL; DR

- Synthetic images from a state-of-the-art text-to-image generator, though ***not as good as real images***, can be useful sources for image recognition.

# Motivation

- Recent text-to-image models demonstrate incredible sample quality and diversity - can they replace *real* data for deep learning model training? (very plausible curiosity)

# Evaluations

### Zero-shot

- Train the classification head with the CLIP encoder *frozen*.
- Generate 2000 images for each class (downstream).
- Upon the baseline that naively uses class label as the text prompt (for GLIDE), introduce Language Enhancement (**LE**) that enriches the caption with T5 word-to-sentence (*e.g., airplane -> a white airplane hover- ing over a beach and a city*).
- As **LE** can inject additional noise to the synthetic data distribution, use CLIP Filter (**CF**) that removes low-confidence synthetic samples based on CLIP embedding similarity.
- Additional Soft Cross-Entropy (**SCE**) strategy strengthens the robustness of the framework.
- Diversity (**LE**) and reliability (**CF+SCE**) combined yields the best performance.

<img width="954" alt="image" src="https://user-images.githubusercontent.com/46308407/207529862-c9eafac2-00d5-43a1-bf42-7f7912702440.png">
<img width="989" alt="image" src="https://user-images.githubusercontent.com/46308407/207529892-5099335b-5ef3-4069-9b99-62196bdc9b8c.png">

- Additional experiments reveal that unfreezing CLIP image encoder hurts its recognition capacity.
- Also, when training from scratch, synthetic samples are much less sample-efficient, possibly due to the domain gap w.r.t. the test set images.

<img width="936" alt="image" src="https://user-images.githubusercontent.com/46308407/207530331-5a0bcd55-015c-418f-8f60-bdab63c79bd5.png">

### Few-shot

- In order to maximally utilize the few real images, the authors propose two strategies: (1) Real Filtering (**RF**) : abandons confusing samples (2) Real Guidance (**RG**) : start the generative diffusion with noised real images (minimize visual discrepancy).
- Generate 800 images per class to finetune the model.
- Additional synthetic samples provide performance boost, with expected diminishing gain w.r.t. the real set size.

<img width="973" alt="image" src="https://user-images.githubusercontent.com/46308407/207531300-8b364591-d0b7-4855-823d-8167a3477c13.png">
<img width="975" alt="image" src="https://user-images.githubusercontent.com/46308407/207531334-95694df9-1d44-4257-88e5-8ad3f334e038.png">

- Freezing the BN stats is beneficial since its not accurate (or stable) to obtain them from synthetic (or few real) samples.
- Mixed training (*I guess pooling the real+synthetic samples during training, perhaps with upsampling weight on the real*) works better than two-phase training (by regularizing the domain gap between the two, per the authors).

### Transferability (pre-training)

- Using real images (such as ImageNet) works better, but with 2x scale, models trained with synthetic data performs on par.
- Plus, using a richer label space helps (*e.g., Imagenet-2k instead of Imagenet-1k*).

<img width="853" alt="image" src="https://user-images.githubusercontent.com/46308407/207531831-5e75bd96-1caf-4a3b-8dc6-b76d94f36146.png">
<img width="977" alt="image" src="https://user-images.githubusercontent.com/46308407/207531929-e5090383-0a4a-4a16-aa9b-530f8f28faf2.png">

# Inspiration

- Using T5 word-to-sentence to enrich the label space (*personal memo*)
- Freezing the BN parameters when the statistical estimate might likely be inaccurate (*personal memo*)
- Real Guidance effectively diffuses a real sample to the latent space and perform deformation (stochastic generative process). Wonder if this can be better replaced with recent techniques like reconstruction guidance (#183 )
- May further incorporate better data generation strategy, especially for contrastive pretraining (e.g., by controlling the distance in latent space)
